server/src/entity/chunk.rs

```diff
@@ -155,13 +159,79 @@ impl<'a> System<'a> for ChunkEntityUpdateSystem {
     }
 }
 
+/// System for spawning entities inside newly-loaded chunks.
+///
+/// This system listens to `ChunkLoadEvent`s.
+#[derive(Default)]
+pub struct EntityChunkLoadSystem {
+    reader: Option<ReaderId<ChunkLoadEvent>>,
+}
+
+impl<'a> System<'a> for EntityChunkLoadSystem {
+    type SystemData = (Read<'a, EventChannel<ChunkLoadEvent>>, Read<'a, Util>);
+
+    fn run(&mut self, data: Self::SystemData) {
+        let (load_events, util) = data;
+
+        for event in load_events.read(self.reader.as_mut().unwrap()) {
+            for entity in &event.entities {
+                match entity {
+                    EntityData::Item(item_data) => {
+                        let pos = item_data.entity.read_position();
+                        if let Some(pos) = pos {
+                            util.spawn_item(
+                                pos,
+                                item_data
+                                    .entity
+                                    .read_velocity()
+                                    .unwrap_or_else(|| glm::vec3(0.0, 0.0, 0.0)),
+                                ItemStack::new(
+                                    Item::from_identifier(item_data.item.item.as_str())
+                                        .unwrap_or(Item::Stone),
+                                    item_data.item.count,
+                                ),
+                            )
+                        }
+                    }
+                    EntityData::Arrow(arrow_data) => {
+                        let pos = arrow_data.entity.read_position();
+                        if let Some(pos) = pos {
+                            util.spawn_arrow(
+                                pos,
+                                arrow_data
+                                    .entity
+                                    .read_velocity()
+                                    .unwrap_or_else(|| glm::vec3(0.0, 0.0, 0.0)),
+                                arrow_data.critical > 0,
+                                None, // TODO: Load shooter UUID
+                            );
+                        }
+                    }
+                    // TODO: Spawn remaining entity types here.
+                    EntityData::Unknown => {
+                        trace!("Chunk {:?} contains an unknown entity type", event.pos);
+                    }
+                }
+            }
+        }
+    }
+
+    fn setup(&mut self, world: &mut World) {
```

Comment:
There's a macro for implementing the setup function for event readers. `setup_impl!(reader)` should do the trick here.